PublishReleaseInstructions.md

```diff
@@ -0,0 +1,35 @@
+## Steps to release and push out a new update.
+
+1. Change program Assembly info inside of Visual Studios.
+	* This is located in FeBuddyWinFormUI under Properties
+2. Change Program version info inside of GlobalConfig.
+	* This is located in FeBuddyLibrary
+3. Clean Solution, Build Solution
+	* Be sure Releases is checked and NOT debug.
+4. Publish program with Visual Studio
+	* With FeBuddyWinFormUI selected, go to Build > Publish Selection in the top menu
+    * Create a new/use an existing Folder profile with the following settings
+        + Configuration - Release | Any CPU
+        + Target Framework - net6.0-windows
+        + Deployment Mode - Self-contained
+        + Target Runtime - win-x86
+        + Produce Single File - Yes
+        + Enable ReadyToRun Compression - No
+    * Publish using this profile
+5. Pack using Squirrel
+    * Download and extract the ZIP archive from the latest release of [Clowd.Squirrel](https://github.com/clowd/clowd.squirrel)
+    * Move the entire `FeBuddyWinFormUI\bin\Release\net6.0-windows\publish\Releases\win-x86` to the folder you extracted the ZIP archive in
+    * Run the following command in PowerShell while your working directory is the folder you extracted the ZIP archive in
+        + `.\Squirrel pack --packName "FE-BUDDY" --packVersion "1.1.0" --packAuthors "Kyle Sanders" --packDirectory win-x86`. I have the author set to Kyle Sanders because of the current Start Menu folder.
```

Comment:
I am not sure why this is set to "Kyle Sanders". Can you have multiple authors in here?

Reply:
At the moment, FE-BUDDY installs a shortcut to `C:\ProgramData\Microsoft\Windows\Start Menu\Kyle Sanders`.  I don't know why that happens, but I put Kyle in the instructions so it would put the shortcut in the same folder to avoid duplicates.

Reply:
Wait so if I was to run setup.exe on my computer, it would create that specific folder `C:\ProgramData\Microsoft\Windows\Start Menu\Kyle Sanders` on my computer? 

Do you have any idea on how to make it say... `C:\ProgramData\Microsoft\Windows\Start Menu\FE-Buddy`

Reply:
Probably the easiest thing to do would be to leave the author argument as "Kyle Sanders" when you build your release, and then add an onUpgrade hook to `HandleEvents` that removes the old shortcut and adds a new shortcut in the desired locations. I am not at visual studio, but it would look roughly like: 

```cs
SquirrelAwareApp.HandleEvents(onUpdate: (v) => {
    using var um = new UpdateManager(...);
    // remove the "Programs/Kyle Sanders/myapp.lnk" shortcut
    um.RemoveShortcutsForThisExe(ShortcutLocations.StartMenu); 
    // add a new shortcut, in the start menu root without a folder and on the desktop
    um.AddShortcutsForThisExe(ShortcutLocations.StartMenuRoot | ShortcutLocations.Desktop);
});
```

These hooks get run when the app is initially installed, when it is updated to a new version, and when it is uninstalled; Don't forget to remove any shortcuts you create in the uninstall hook too!

Reply:
Thanks for the suggestion! I assume we'll just have to stay with Kyle as the author in the future as well? I don't see a specific entry in the documentation for `RemoveShortcutsForThisExe()`, and looking in VS's definition it would seem it doesn't accept absolute paths. Is there anything stopping me from just deleting the file if it exists?

Reply:
Yes, that's a good point. If you update the author argument, create the desired shortcuts in the installed and upgrade hooks, and add a check to delete the old shortcut and Kyle Sanders folder, this should work fine.

Reply:
Alright. I've pushed something for that, it deletes the Kyle Sanders folder and adds a shortcut in the root. What should the author be in the instructions, @Nikolai558 ?

Reply:
@wiggleforlife - In My opinion, the author should honestly be the program name "FE-Buddy" since (at least the way I understand it) there cannot be multiple Authors listed. Also, if the 'Author' is directly linked to the 'shortcut' File/Dir then it really should be the program name.